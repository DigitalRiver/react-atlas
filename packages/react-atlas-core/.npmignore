.*rc
.eslintignore
.editorconfig
webpack.config.*
jsconfig.json
CONTRIBUTING.md
CODE_OF_CONDUCT.md
src
webpack
