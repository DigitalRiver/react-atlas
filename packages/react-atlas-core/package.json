{
  "name": "react-atlas-core",
  "publishConfig": {
    "registry": "http://registry.npmjs.org/"
  },
  "version": "0.1.0-beta27",
  "description": "Core set of composable components for react-atlas",
  "author": "Digital River Experience Engineering (https://github.com/orgs/DigitalRiver/teams/experience-engineering)",
  "main": "./lib/index.js",
  "bugs": {
    "url": "https://github.com/DigitalRiver/react-atlas/issues"
  },
  "keywords": [
    "react",
    "react-component",
    "atlas",
    "CSS Modules",
    "components"
  ],
  "license": "MIT",
  "dependencies": {
    "classnames": "^2.2.5",
    "inputmask-core": "^2.2.0",
    "light-table": "0.1.1",
    "moment": "^2.19.1",
    "proptypes": "^1.0.0",
    "react": "^15.6.2",
    "react-datepicker": "^0.60.2",
    "react-dom": "^15.6.2",
    "react-dropzone": "^4.2.1"
  },
  "devDependencies": {
    "babel-core": "^6.24.1",
    "babel-loader": "^7.0.0",
    "babel-preset-es2015": "^6.24.1",
    "babel-preset-react": "^6.24.1",
    "css-loader": "^0.28.1",
    "style-loader": "^0.19.0",
    "webpack": "^3.8.1"
  },
  "repository": "https://github.com/DigitalRiver/react-atlas/tree/master/packages/react-atlas-core"
}
