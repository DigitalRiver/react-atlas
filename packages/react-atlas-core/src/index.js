export { AccordionCore } from "./Accordion";
export { AvatarCore } from "./Avatar";
export { ButtonCore } from "./Button";
export { CardCore } from "./Card";
export { CheckboxCore } from "./Checkbox";
export { CheckboxGroupCore } from "./CheckboxGroup";
export { DatePickerCore } from "./DatePicker";
export { DialogCore } from "./Dialog";
export { DropdownCore } from "./Dropdown";
export { FormCore } from "./Form";
export { FileUploadCore } from "./FileUpload";
export { HintCore } from "./Hint";
export { InputCore } from "./Input";
export { ModalCore } from "./Modal";
export { PanelCore } from "./Panel";
export { ProgressBarCore } from "./ProgressBar";
export { RadioCore } from "./Radio";
export { RadioGroupCore } from "./RadioGroup";
export { SwitchCore } from "./Switch";
export { TabCore } from "./Tab";
export { TableCore } from "./Table";
export { TableHeaderCore } from "./TableHeader";
export { TabListCore } from "./TabList";
export { TabPanelCore } from "./TabPanel";
export { TabsCore } from "./Tabs";
export { TaskCore } from "./Task";
export { TaskbarCore } from "./Taskbar";
export { TextAreaCore } from "./TextArea";
export { TextFieldCore } from "./TextField";
export { TimerCore } from "./Timer";
export { TooltipCore } from "./Tooltip";
