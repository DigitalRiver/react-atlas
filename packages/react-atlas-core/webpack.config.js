var path = require('path');
var webpack = require('webpack');

let config = {
  entry: [
    './src/index.js'
  ],
  output: {
    filename: 'index.js',
    path: path.join(__dirname, 'lib'),
    publicPath: '/lib/',
    libraryTarget: 'commonjs2'
  },
  externals: [{
      'react': {
          root: 'React',
          commonjs2: 'react',
          commonjs: 'react',
          amd: 'react'
      }
  }, {
      'react-dom': {
          root: 'ReactDOM',
          commonjs2: 'react-dom',
          commonjs: 'react-dom',
          amd: 'react-dom'
      }
  }],
  module: {
    rules: [
      {
        test: /\.css$/,
        use: [
          'style-loader',
          {
            loader: 'css-loader',
            options: { modules: true }
          },
        ],
      },
      {
        test: /\.js$/,
        exclude: /(node_modules|bower_components)/,
        use: {
          loader: 'babel-loader',
          query: {compact: true}
        }
      }
    ],
  },
  plugins: [
    new webpack.DefinePlugin({
      'process.env': {
        'NODE_ENV': JSON.stringify(process.env.NODE_ENV)
      }
    })
  ]
};

if(process.env.NODE_ENV === "production") {
  config.plugins.push(new webpack.optimize.UglifyJsPlugin({
    compress: {
      warnings: false
    }
  }))
  config.plugins.push(new webpack.optimize.AggressiveMergingPlugin());
}

module.exports = function() {
  return config;
}