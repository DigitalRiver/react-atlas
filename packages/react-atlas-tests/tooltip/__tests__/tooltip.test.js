import React from "react";
import { mount } from "enzyme";
import { TooltipCore } from "react-atlas-core";

const message = "The tooltip message";
const inline = false;
const delay = 10;
const hide = true;
const position = "top";

describe("Testing tooltip component", () => {
  it("Set props should match what was passed in", function() {
    const result = mount(
      <TooltipCore
        tooltip={message}
        inline={inline}
        tooltipDelay={delay}
        tooltipHideOnClick={hide}
        position={position}
      />
    );
    expect(result.props().tooltip).toBe(message);
    expect(result.props().inline).toBe(inline);
    expect(result.props().tooltipDelay).toBe(delay);
    expect(result.props().tooltipHideOnClick).toBe(hide);
    expect(result.props().position).toBe(position);
  });
});
