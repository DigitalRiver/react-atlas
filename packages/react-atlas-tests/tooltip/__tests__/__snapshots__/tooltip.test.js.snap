// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tooltip component render Render correctly 1`] = `
<div
  className="class"
  data-tooltip=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="tooltip"
  style={undefined}
  styleName="block tooltip tooltip-top"
>
  <i
    className="<i class='fa fa-github'></i>"
  />
  
</div>
`;
